src/OpenCensus.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,131 @@
+﻿// <copyright file="HttpHandlerDiagnosticListener.cs" company="OpenCensus Authors">
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of theLicense at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenCensus.Collector.Dependencies.Implementation
+{
+    using System;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using OpenCensus.Trace;
+
+    internal class HttpHandlerDiagnosticListener : ListenerHandler
+    {
+        private readonly PropertyFetcher startRequestFetcher = new PropertyFetcher("Request");
+        private readonly PropertyFetcher stopResponseFetcher = new PropertyFetcher("Response");
+        private readonly PropertyFetcher stopExceptionFetcher = new PropertyFetcher("Exception");
+        private readonly PropertyFetcher stopRequestStatusFetcher = new PropertyFetcher("RequestTaskStatus");
+
+        public HttpHandlerDiagnosticListener(ITracer tracer, ISampler sampler) : base("HttpHandlerDiagnosticListener", tracer, sampler)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (!(this.startRequestFetcher.Fetch(payload) is HttpRequestMessage request))
```

Comment:
the cost of having a bug here (in any callback) is extremely high. So if we accidentally throw, it will break user app and prevent the http call. We should put all callbacks in try/catch and log exceptions.

Reply:
It is handled in `DiagnosticSourceListener`